library/core/Cargo.toml

```diff
@@ -29,7 +29,6 @@ debug_typeid = []
 [lints.rust.unexpected_cfgs]
 level = "warn"
 check-cfg = [
-    'cfg(bootstrap)',
```

Comment:
Despite this removal, cfg(bootstrap) gets accepted in libcore apparently. However, it now gets rejected for out-of-tree builds, such as https://github.com/rust-lang/miri-test-libstd/. That's bad since it means we'll land changes that then fail to build elsewhere.

I think either the cfg(bootstrap) here should be added back, or our CI needs to ensure that there truly isn't any cfg(bootstrap) in libcore.

Also see https://rust-lang.zulipchat.com/#narrow/channel/269128-miri/topic/Miri.20test-libstd.20Failure.20.282025-06.29/with/522715612